components/Footer.jsx

```diff
@@ -0,0 +1,219 @@
+                  <a
+                    target="_blank"
+                    rel="noopener noreferrer"
+                    href="https://github.com/mastodon/mastodon/discussions"
+                  >
+                    <FormattedMessage
+                      id="nav.support"
+                      defaultMessage="Support"
+                    />
+                  </a>
+                </li>
+              </ul>
+            </div>
+            <div className="Footer__menu">
+              <h4 className="Footer__menu-heading h6">
+                <FormattedMessage id="nav.company" defaultMessage="Company" />
+              </h4>
+              <ul className="b2">
+                <li>
+                  <Link href="/imprint">
+                    <a>
+                      <FormattedMessage
+                        id="contact_us"
+                        defaultMessage="Contact us"
+                      />
+                    </a>
+                  </Link>
+                </li>
+                <li>
+                  <a
+                    target="_blank"
+                    rel="noopener noreferrer"
+                    href="https://blog.joinmastodon.org/"
+                  >
+                    <FormattedMessage id="nav.blog" defaultMessage="Blog" />
+                  </a>
+                </li>
+                <li>
+                  <a href="/press-kit.zip">
+                    <FormattedMessage
+                      id="credits.press_kit"
+                      defaultMessage="Press kit"
+                    />
+                  </a>
+                </li>
+                <li>
+                  <Link href="/privacy-policy">
+                    <a>
+                      <FormattedMessage
+                        id="privacy_policy"
+                        defaultMessage="Privacy policy"
+                      />
+                    </a>
+                  </Link>
+                </li>
+              </ul>
+            </div>
+          </div>
+        </div>
+      </nav>
+    </footer>
+    <style jsx>{`
+      .Footer {
+        background: var(--eggplant);
+        color: var(--white);
+        padding: 6rem 0;
+        text-align: center;
+      }
+      .Footer__menu-heading {
+        line-height: 1.5rem;
+        padding: 0.5rem 0;
+      }
+      .Footer__grid {
+        row-gap: 2.5rem;
+      }
+      .Footer__menu ul {
+        display: flex;
+        flex-direction: column;
+        gap: 0.5rem;
+        list-style: none;
+        padding: 0;
+        margin: 0;
+      }
+      .Footer__menus {
+        display: grid;
+        gap: 2.5rem;
+      }
+      .Footer__menu {
+        display: flex;
+        flex-direction: column;
+        gap: 0.5rem;
+      }
+      .Footer__menu a {
+        display: inline-block;
+        padding-block: 0.5rem;
+      }
+      .Footer__logo {
+        width: 5.5rem;
+        max-width: 100%;
+        justify-self: center;
+      }
+      @media (min-width: 576px) {
```

Comment:
media queries feels like a maintenance pain point—open to suggestions or workarounds

- we could add the sass plugin for styled-jsx, but i think [it'd be an SWC opt-out](https://github.com/Thream/styled-jsx-plugin-sass/issues/100)... by the same token, it [looks (deceptively?) simple to write the plugin](https://github.com/swc-project/plugins/blob/main/packages/styled-jsx/src/lib.rs). sadly, it seems like the maintainers have been [viewing](https://twitter.com/timneutkens/status/1430211399150710786) styled-jsx and particularly the plugins as secondary/legacy tooling of late
- we could put this in global scss, which is bulky but reliable
- we could use [css modules](https://nextjs.org/docs/basic-features/built-in-css-support#adding-component-level-css) (i just usually find the DX of kinda-sorta-colocating, importing, and drilling into the css import cumbersome)
- tailwind might be a better candidate here in react than in our siteleaf environment now that we have more runtime control over state
- hopefully in a year or two [@custom-media](https://www.w3.org/TR/mediaqueries-5/#custom-mq) will help 🙏😩 

the `rem()` function is the other thing i find missing in plain css, but in general the mockup is well-unitized so all of the rem values i'm writing are in increments of 0.5 or 1.0

Reply:
oo right....initial thoughts:
- I'll say no to putting it in global scss, since if we want to be consistent with the media queries variables/mixins, every responsive style will need to be in there
- also no to css modules, since i also find it cumbersome 😓 

are there any other css-in-js options that work with SWC? If styled-jsx is already considered legacy tooling, maybe we should opt for something else now?

Reply:
if not, I'm also down to use tailwind!